### ISSUE TITLE
Moscow Prayers Times

### ISSUE THREAD

**Issue by yangar** [zulfahmi93/prayer-times#1]
Some couples of locations return an ArgumentOutOfRangeException.

Exemple for Moscow:
```
var when = Instant.FromUtc(2018, 7, 2, 0, 0);
// Init settings.
var settings = new PrayerCalculationSettings();
// Set calculation method to JAKIM (Fajr: 18.0 and Isha: 20.0).
settings.CalculationMethod.SetCalculationMethodPreset(when, CalculationMethodPreset.DepartmentOfIslamicAdvancementOfMalaysia);
// Init location info.
var geo = new Geocoordinate(55.75, 37.61, 0);
```
```
ArgumentOutOfRangeException: Value should be in range [-402653184-402653183]
Parameter name: hours
Actual value was NaN.
```

**Comment by zulfahmi93** [comment]
Thanks for the report. I have investigated the problem and working on the solution. This may take some time as the problem only occur at location with a higher latitude.

**Comment by yangar** [comment]
Yes look like the problem occur in the **ComputeSolarTime** calculaton  on AstronomyMath.cs.
I ve tried to fix it using an other library : https://github.com/porrey/Solar-Calculator
Without success, my knowledge on this topic is quiet limited.

An other interesting libraiy:
https://github.com/xeqlol/SolarPositionAlgorithm
Thank you for your work!

**Comment by highwingers** [comment]
Any work around this? its also throwing same error for Manchester UK as well. Thank you

**Comment by highwingers** [comment]
> Thanks for the report. I have investigated the problem and working on the solution. This may take some time as the problem only occur at location with a higher latitude.

Any Update?

**Comment by highwingers** [comment]
> Yes look like the problem occur in the **ComputeSolarTime** calculaton on AstronomyMath.cs.
> I ve tried to fix it using an other library : https://github.com/porrey/Solar-Calculator
> Without success, my knowledge on this topic is quiet limited.
> 
> An other interesting libraiy:
> https://github.com/xeqlol/SolarPositionAlgorithm
> Thank you for your work!

did you find any work around?

**Comment by snice1981** [comment]
Same problem here. Someone managed to find a solution, because the author seems to not maintain this code anymore.

**Comment by highwingers** [comment]
> Same problem here. Someone managed to find a solution, because the author seems to not maintain this code anymore.

did anyone find the solution?
